client/src/partials/purchases/modals/search.modal.js

```diff
@@ -0,0 +1,58 @@
+angular.module('bhima.controllers')
+.controller('SearchPurchaseOrderModalController', SearchPurchaseOrderModalController);
+
+// dependencies injections
+SearchPurchaseOrderModalController.$inject = [
+  'UserService', 'SupplierService', 'NotifyService', '$uibModalInstance', 'SearchFilterFormatService'
+];
+
+function SearchPurchaseOrderModalController(Users, Suppliers, Notify, Instance, SearchFilterFormat) {
+  var vm = this;
+
+  // gloabal variables 
+  var noMissingDatePart;
+
+  // global methods
+  vm.validate = validate;
+  vm.cancel = Instance.close;
+  vm.submit = submit;
+
+  // init 
+  init();
+
+  // load users 
+  Users.read(null, { detailled: 1 })
```

Comment:
Do you need detailed information about the user?  I think you can get away with `Users.read()`.